### PR TITLE
To add Rprop documentation

### PR DESCRIPTION
**Pull request by iramazanli**
It has been discussed before that adding description of Optimization algorithms to PyTorch Core documentation may result in a nice Optimization research tutorial. In the following tracking issue we mentioned about all the necessary algorithms and links to the originally published paper  https://github.com/pytorch/pytorch/issues/63236.

In this PR we are adding description of Rprop to the documentation.  For more details, we refer to the paper  http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.21.1417

Rpropalg

### CODE REVIEW

File: torch/optim/rprop.py [hunk 1]
```diff
@@ -4,7 +4,37 @@
...
 class Rprop(Optimizer):
-    """Implements the resilient backpropagation algorithm.
+    r"""Implements the resilient backpropagation algorithm.
+
+    .. math::
+       \begin{aligned}
+            &\rule{110mm}{0.4pt}                                                                 \\
+            &\textbf{input}      : \theta_0 \in \mathbf{R}^d \text{ (params)},f(\theta)
+                \text{ (objective)},                                                             \\
+            &\hspace{13mm}      \eta_{+/-} \text{ (etaplus, etaminus)}, \Gamma_{plus/minus}
```
Comment (albanD) [inline torch/optim/rprop.py:14]: This is missing initialization for `g_{prev}` no?

File: torch/optim/rprop.py [hunk 2]
```diff
@@ -4,7 +4,37 @@
...
+                \text{ (boundaries for lr)}                                               \\[-1.ex]
```
Comment (albanD) [inline torch/optim/rprop.py:15]: Is that the "step_sizes" argumeents?

File: torch/optim/rprop.py [hunk 3]
```diff
@@ -4,7 +4,37 @@
...
+            &\rule{110mm}{0.4pt}                                                                 \\
```
Comment (albanD) [inline torch/optim/rprop.py:16]: Also `\eta_t` initialization is missing no?

**Review by albanD — COMMENTED** [review]
The code is not equivalent if lr is not inside step_sizes range. Is that a problem?

File: torch/optim/rprop.py (hunk 1 above)
Comment (iramazanli) [reply torch/optim/rprop.py:14]: done

File: torch/optim/rprop.py (hunk 2 above)
Comment (iramazanli) [reply torch/optim/rprop.py:15]: yes, exactly !

File: torch/optim/rprop.py (hunk 3 above)
Comment (iramazanli) [reply torch/optim/rprop.py:18]: done !

File: torch/optim/rprop.py (hunk 2 above)
Comment (albanD) [reply torch/optim/rprop.py]: It should be mentioned explicitly then.

File: torch/optim/rprop.py [hunk 4]
```diff
@@ -4,7 +4,39 @@
...
+            &\textbf{input}      : \theta_0 \in \mathbf{R}^d \text{ (params)},f(\theta)
+                \text{ (objective)},                                                             \\
+            &\hspace{13mm}      \eta_{+/-} \text{ (etaplus, etaminus)}, \Gamma_{max/min}
+                \text{ (boundaries for lr)}                                                      \\
+            &\textbf{initialize} :   g^0_{prev} \leftarrow 0,
+                \: \eta_0 \leftarrow \text{lr (learning rate)}                                   \\
+            &\rule{110mm}{0.4pt}                                                                 \\
+            &\textbf{for} \: t=1 \: \textbf{to} \: \ldots \: \textbf{do}                         \\
+            &\hspace{5mm}g_t           \leftarrow   \nabla_{\theta} f_t (\theta_{t-1})           \\
+            &\hspace{5mm} \textbf{for} \text{  } i = 0, 1, \ldots, d-1 \: \mathbf{do}            \\
```
Comment (albanD) [inline torch/optim/rprop.py:21]: I missed it on first read but d is not defined?

Comment (iramazanli) [reply torch/optim/rprop.py:21]: d is defined actually implicitly, if you see, input part of the algorithm `\theta \in R^d`

Comment (albanD) [reply torch/optim/rprop.py:21]: Ho ok, missed that :D

File: torch/optim/rprop.py (hunk 2 above)
Comment (iramazanli) [reply torch/optim/rprop.py]: done!

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**